streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsConventions.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals.metrics;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public final class StreamsMetricsConventions {
+    private StreamsMetricsConventions() {
+    }
+
+    public static String threadLevelSensorName(final String threadName, final String sensorName) {
```

Comment:
Similar to my other comment on `CommonStreamsMetrics`: could we move these static functions to the corresponding level metrics class?

Reply:
I thought it would be nice if the metrics naming conventions were all in one place, to help us maintain consistency. Right now, we have one (internal) naming convention enforced via our (public) metrics API: `StreamsMetrics`, but we also have a bunch of metrics with no defined naming convention declared, e.g., in `StreamsMetricsThreadImpl`.

I think it'll be simpler for people to use the metrics if we maintain consistency throughout the whole streams code-base, and it'll be simpler to maintain consistency if we keep all the naming conventions in one file.

Reply:
Cool.